src/components/main/codeView/CodeView.tsx

```diff
@@ -72,36 +72,38 @@ export default function CodeView(props: CodeViewProps) {
 
   const isFirst = useRef<boolean>(true);
 
-  const codeContent = useRef<string>("");
+  const [codeContent, setCodeContent] = useState<string>("");
```

Comment:
How did changing this helped? Curious to know!

Reply:
in perspective of updates React is more useful with states, now it more clear to use. Additionally the main purpose of Refs in react is direct manipulation with virtual DOM elements, not to handle data changes.

Reply:
useState or useReducer is useful when we want some value that updates on the UI when we change it.

While useRef can hold ref of dom elements but also can store anything inside the `current`. It's useful when we want to access the value immediately without having a useEffect to find out if it's actually updated.